src/main/java/io/r2dbc/postgresql/PostgresqlConnectionConfiguration.java

```diff
@@ -60,6 +62,8 @@
 
     private final List<Extension> extensions;
 
+    private final int fetchSize;
```

Comment:
How about turning this into a `ToIntFunction<String>`?  The argument would be the actual SQL. That way, we can have a default fetchsize while being able to react to specific SQL statements to return a different fetch size.